application/src/main/java/org/opentripplanner/service/streetdecorator/OsmStreetDecoratorRepository.java

```diff
@@ -0,0 +1,23 @@
+package org.opentripplanner.service.streetdecorator;
+
+import java.io.Serializable;
+import java.util.Optional;
+import org.opentripplanner.service.streetdecorator.model.EdgeLevelInfo;
+import org.opentripplanner.street.model.edge.Edge;
+
+/**
+ * Store OSM level and incline data used for returning responses to requests.
+ * <p>
+ * This is a repository to support the {@link OsmStreetDecoratorService}.
+ */
+public interface OsmStreetDecoratorRepository extends Serializable {
+  /**
+   * Associate the edge with level information.
+   */
+  void addEdgeLevelInformation(Edge edge, EdgeLevelInfo edgeLevelInfo);
```

Comment:
An Edge is internal model class in the street model, if an edge is modified this reference will no longer be valid. We should use an edge id here. This is referencing an entity in another agregate.